### PR TITLE
Refactor atop rapcom

### PR DESCRIPTION
**Pull request by zancas**
@dangle 
Fixes: #42 

This greatly simplifies the code.    Note I haven't created a `contain` cli utility.

### CODE REVIEW

**Comment by zancas** [comment]
@dangle waddya think?

File: tox.ini [hunk 1]
```diff
@@ -19,22 +19,22 @@ deps =
     test: pytest-cov
     test: coveralls
     lint: pep8
-    lint: pep257
+#lint: pep257
```
Comment (dangle) [inline tox.ini]: Why comment these out? Merges don't require a passing build yet, do they?

File: containenv/commands/containenv.py [hunk 2]
```diff
@@ -1,5 +1,38 @@
-#!/usr/bin/env python
+"""
+Usage: containenv [--verbose] [--debug <LEVEL>] ( init | help ) [<args>...]
```
Comment (dangle) [inline containenv/commands/containenv.py]: If you take `[--log-level <level>]` as an option you get logging for free from rapcom.

Comment (dangle) [inline containenv/commands/containenv.py]: So this is only used in the case of `containenv help`?

File: containenv/commands/containenv.py [hunk 3]
```diff
@@ -1,5 +1,38 @@
...
+from pkg_resources import resource_filename
+from rapcom.display import run_tasks
+
+from ..config.dependencymaps import EXTENSION_MAP
+from ..config.dependencymaps import FILENAME_MAP
+from ..config.dependencymaps import DIRNAME_MAP
+from ..exceptions import ProjectAlreadyInitialized
+from ..exceptions import ProjectDirectoryDoesNotExist
+from ..exceptions import ProjectContainsNoRegisteredNodes
+_DEFAULT_DOC = __doc__.format("""Common containenv commands:
```
Comment (dangle) [inline containenv/commands/containenv.py]: I'll try to get support for this into `rapcom` for 0.1.2.

File: containenv/commands/containenv.py [hunk 4]
```diff
@@ -15,36 +48,17 @@
...
-from containenv.display import run_tasks
-from containenv.exceptions import ProjectAlreadyInitialized
-from containenv.exceptions import ProjectDirectoryDoesNotExist
-from containenv.exceptions import ProjectContainsNoRegisteredNodes
+logger = logging.getLogger(__name__)
 
 
-logger = logging.getLogger(__name__)
+class Init(object):
+    '''Usage: containenv init [metasource]
```
Comment (dangle) [inline containenv/commands/containenv.py]: You should add an `Arguments` section to explain `[metasource]`.

File: containenv/commands/containenv.py [hunk 5]
```diff
@@ -219,3 +234,41 @@ def write_runcontainenv(self):
     def publish_image(self):
         '''if there's a shared registry publish to it'''
         pass
+
+
+class Activate(object):
+    '''Usage: containenv activate [help]
```
Comment (dangle) [inline containenv/commands/containenv.py]: Does this have weird indentation when you run `containenv activate -h`? I wouldn't expect `dedent` to do anything here.

Comment (dangle) [inline containenv/commands/containenv.py]: What does `containenv activate help` do? The built in subcommand help is `containenv help activate`. Custom help message?

File: containenv/commands/containenv.py [hunk 6]
```diff
@@ -219,3 +234,41 @@ def write_runcontainenv(self):
...
+
+    def __call__(self):
+        """Write container config and build container .
+
+        Returns:
+            The exit code for the command, or None, implying a successful run.
+            run runcontainenv.sh
+        """
+        logging.debug('target is {}'.format(self.target))
+        subprocess.call(["bash", self.target])
```
Comment (dangle) [inline containenv/commands/containenv.py]: This should return the result of `subprocess.call`.

File: containenv/commands/test/test_containenv_init.py [hunk 7]
```diff
@@ -19,10 +19,10 @@
```
Comment (dangle) [inline containenv/commands/test/test_containenv_init.py]: These inline tests should be moved out of the packaged code.

**Review by dangle — CHANGES_REQUESTED** [review]
Looks pretty good. Fix the return value on the subprocess call and look at that indentation, then I think it's ready.

File: containenv/commands/test/test_containenv_init.py (hunk 7 above)
Comment (zancas) [inline containenv/commands/test/test_containenv_init.py]: Oh dear!   I disagree with this...   why exclude tests?

File: tox.ini (hunk 1 above)
Comment (zancas) [reply tox.ini]: uncommented

File: containenv/commands/containenv.py (hunk 2 above)
Comment (zancas) [reply containenv/commands/containenv.py]: `--verbose`excised.

Comment (zancas) [reply containenv/commands/containenv.py]: Removed, depends on the `rapcom` defaults for the primary command.

File: containenv/commands/containenv.py (hunk 3 above)
Comment (zancas) [reply containenv/commands/containenv.py]: Nice to have...   I think validation is much more important.

File: containenv/commands/containenv.py (hunk 4 above)
Comment (zancas) [reply containenv/commands/containenv.py]: Added.

File: containenv/commands/containenv.py (hunk 5 above)
Comment (zancas) [reply containenv/commands/containenv.py]: It was indented...   I fixed it.

Comment (zancas) [reply containenv/commands/containenv.py]: It's gone.

File: containenv/commands/test/test_containenv_init.py (hunk 7 above)
Comment (zancas) [reply containenv/commands/test/test_containenv_init.py]: done

File: containenv/commands/containenv.py (hunk 5 above)
Comment (zancas) [reply containenv/commands/containenv.py]: oops...  this breaks style checks...  and upon a moments reflection...   properly so...   I think rapcom has to handle this gracefully.

**Review by dangle — APPROVED** [review]

**Pull request merged**